scripts/release.js

```diff
@@ -125,6 +125,11 @@ async function main() {
   step('\nUpdating package version...')
   updateVersion(targetVersion)
 
+  if (pkg.scripts.lint != null) {
+    step('\nLint package...')
+    await run('yarn', ['lint'])
+  }
```

Comment:
@Shinigami92 we discussed this with Anthony and it is better to avoid changing the release process. Isn't linting already being checked during prebuild anyways and in the CI? In any case, I think it would be good that we are able to merge this so if you think this is important maybe we could discuss it in a separate PR.

Reply:
Yeah, thinking we can just remove this addition to the release script and the rest LGTM.

Reply:
I added this especially here, cause I removed it from the `package.json`'s `prebuild`!
Also (I don't know why), but in special cases the lint of `packages/vite:lint` catches some other lint as the mono-root lint. That may have to do something with building the `dist` folder and checking if it was build so it can be `required`.

It needs to be removed from the `prebuild` step, cause it would always be triggered in the CI and therefore trigger linting multiple times (see https://github.com/vitejs/vite/pull/2988#issuecomment-827591060)

And yeah, linting is not that relevant as tests, but it safes a bit more stability
If I understand it correctly, if the lint fails here (exit code 1), then only the version number in `package.json` would be bumped up, but only locally and it wasn't committed or tagged already to somewhere. So this is not a problem at all.

---

Should I still remove it?

Reply:
Another option could be to have a ci-build step that doesnt lint

Reply:
Could you explain/propose a bit more? `prebuild` is just called before `build` by npm automatically. That's the main issue here currently.

Reply:
Instead of 
```json
    "prebuild": "rimraf dist && yarn lint",
    "build": "run-s build-bundle build-types",
```
We could have
```json
    "preci-build": "rimraf dist",
    "ci-build": "run-s build-bundle build-types",
    "prebuild": "rimraf dist && yarn lint",
    "build": "run-s build-bundle build-types",
```
And run `ci-build` for the CI workflow?

Reply:
But is it best practice to have two different build scripts for human and CI?
I don't think so :confused: 

Maybe we can check a `CI` environment variable and write an if-else :thinking:  
But also then I would argue: why not the current, bit more human readable solution?

Reply:
I agree with @Shinigami92 - it's already linting now during release, just implicitly, and this PR makes it explicit.

Reply:
But you also changed the way `build` works, no? Before, we were getting the lint step on every build that I think is helpful. Now it is only present independently or during release.

Reply:
It was only there for `release`, so the person who release don't forget to run lint
But linting is done by mono-root `lint` step via CI

Reply:
Sorry for catching up late. This looks good but I am a bit concern about adding more scripts also increases the maintaining surface. And I kinda think, since we are doing scheduled manually releases, we could just have "Make sure all CI passes on the main branch before doing the release" in our releasing SOP, and then we can simply remove the linting in "prebuild". To me, CI results are more reliable than local tests/lints (clean, multiple platforms) as we can't actually cover them all in local env. WDYT?

Reply:
Also thought about this but didn't dare to change the logic 😅
Should we wait on input from Evan here?

Reply:
I think we should keep linting in prebuild not because of the release process, but because it helps when modifying Vite before sending a PR. It is a good idea to me that the build step lints (in the same way that it checks typing) so people can feel more confident before sending a PR. I don't think we should force them to run the lining on the side (nobody would remember). We had this discussion with @Shinigami92, where the focus was the release process, but to me what is important is the regular build process. If you would like to remove linting from release, I think it is fine, but let's avoid removing it from prebuild.
About the `ci-` scripts, that is one way to achieve this, and it is already being used before this PR in `ci-docs` for example.

Reply:
But we have pre commit hooks for this? 🤔

Reply:
Ok, your call then 👍🏼

Reply:
I could try (later in another PR) to use `pre-commit` with `prettier --write` and `eslint` for everything  
Also for `md` and other files, so we have a consistent code base over the whole repo

I think I will come this evening to have a last look into this PR and make a decision

Reply:
I created a task for me to check prettier and eslint: https://github.com/vitejs/vite/issues/3261  
IMO lets just merge this PR for now, so the main issue of multiple occurrences in PR reviews are solved